perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_attribute_def_def_o365EmailAddresses_mu.java

```diff
@@ -89,25 +102,33 @@ public void checkAttributeSemantics(PerunSessionImpl sess, Member member, Attrib
 		}
 		String ucoEmail = UCO + "@muni.cz";
 		if (!emails.contains(ucoEmail)) {
-			throw new WrongAttributeValueException(attribute, member, "does not contain " + ucoEmail);
+			throw new WrongReferenceAttributeValueException(attribute, "does not contain " + ucoEmail);
 		}
 
 		//check for duplicities among all members and groups (urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu)
 		AttributesManagerBl attributesManagerBl = sess.getPerunBl().getAttributesManagerBl();
 		Set<Pair<Integer, Integer>> memberPairs = attributesManagerBl.getPerunBeanIdsForUniqueAttributeValue(sess, attribute);
 		memberPairs.remove(new Pair<>(member.getId(), 0));
 		if (!memberPairs.isEmpty()) {
-			Set<Integer> memberIds = new HashSet<>();
-			for (Pair<Integer, Integer> memberPair : memberPairs) {
-				memberIds.add(memberPair.getLeft());
+			try {
+				Member memberWithDuplicateEmail = sess.getPerunBl().getMembersManagerBl().getMemberById(sess, memberPairs.iterator().next().getLeft());
+				throw new WrongReferenceAttributeValueException(attribute, attribute, memberWithDuplicateEmail, null, "some of the email addresses are already assigned.");
```

Comment:
Where is the original member? You are saying that for this attribute and this member (from check method) there is a problem with value because of this attribute and that member (with duplicate email), right?

Reply:
Oh, sorry, I had it wrong, but hopefully it should be correct now.